File: backend/src/main/java/com/thechair/bookings/entity/Waitlist.java
package com.thechair.bookings.entity;

import com.thechair.salons.entity.Salon;
import com.thechair.services.entity.SalonOffering;
import com.thechair.users.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "waitlists")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Waitlist {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private User customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "salon_id", nullable = false)
    private Salon salon;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "offering_id", nullable = false)
    private SalonOffering offering;

    @Column(name = "preferred_date", nullable = false)
    private LocalDate preferredDate;

    @Column(name = "preferred_time_start")
    private LocalTime preferredTimeStart;

    @Column(name = "preferred_time_end")
    private LocalTime preferredTimeEnd;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private WaitlistStatus status = WaitlistStatus.PENDING;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend-web/src/api/waitlist.ts
import api from './axios'
import { ApiResponse, Waitlist } from '../types'

export const waitlistApi = {
  join: (data: {
    salonId: string
    offeringId: string
    preferredDate: string
    preferredTimeStart?: string
    preferredTimeEnd?: string
  }) => api.post<ApiResponse<Waitlist>>('/customer/waitlist', data).then(r => r.data.data),

  getMyWaitlists: () =>
    api.get<ApiResponse<Waitlist[]>>('/customer/waitlist').then(r => r.data.data),

  leaveWaitlist: (id: string) =>
    api.delete<ApiResponse<void>>(`/customer/waitlist/${id}`).then(r => r.data.data),

  getOwnerWaitlist: () =>
    api.get<ApiResponse<Waitlist[]>>('/owner/waitlist').then(r => r.data.data),
}
